share/ci/pypicongpu_generator.py

```diff
@@ -127,6 +129,50 @@ def get_all_major_pypi_versions(package_name):
     return [str(v) for v in version_map.values()]
 
 
+@typeguard.typechecked
+def combine_requirements(one: pkg_resources.Requirement, two: pkg_resources.Requirement) -> pkg_resources.Requirement:
+    if one.project_name != two.project_name:
+        print(f"requirements for {one.project_name} and {two.project_name} are incomparable!")
+
+    # accumulate all specs
+    specs: list = copy.deepcopy(one.specs)
+    specs.extend(two.specs)
+
+    last_tuple = specs.pop()
+    requirement_string = one.project_name + " " + last_tuple[0] + last_tuple[1]
+    for entry in specs:
+        requirement_string += "," + entry[0] + entry[1]
+
+    return pkg_resources.Requirement(requirement_string)
+
+
+@typeguard.typechecked
+def readInRequirementFiles(requirement_file_names: list[str]) -> dict[str, pkg_resources.Requirement]:
+    total_requirements = {}
+    for requirement_file_name in requirement_file_names:
+        with open(requirement_file_name, "r", encoding="utf-8") as requirement_file:
+            try:
+                # read in file
+                parsed_requirements = pkg_resources.parse_requirements(requirement_file)
+
+                # accumulate
+                for requirement in parsed_requirements:
+                    if requirement.project_name not in total_requirements:
+                        # does not exist yet
+                        total_requirements[requirement.project_name] = requirement
+                    else:
+                        # already exists
+                        total_requirements[requirement.project_name] = combine_requirements(
+                            requirement, total_requirements[requirement.project_name]
+                        )
+
+            except Exception:
+                # ignore all lines, which cannot be parsed
+                # e.g. `-r extra/requirements.txt`
+                pass
```

Comment:
Can these lines be safely ignored and the exception just passed?

Reply:
Yes
If all dependencies required are included in the passed file this never comes up.
If that is not the case, we will get an error in the CI later when we run the tests for code that uses a missing dependency, because it will not be installed in the CI container.